src/ServiceManagerBase.sol

```diff
@@ -135,6 +137,114 @@ abstract contract ServiceManagerBase is ServiceManagerBaseStorage {
+    }
+
+    function _migrateToOperatorSets(uint32[][] memory operatorSetIds, address[] memory operators) internal {
+        AVSDirectory(address(_avsDirectory)).migrateOperatorsToOperatorSets(operators, operatorSetIds);
+    }
+
+    function getOperatorsToMigrate() public view returns (uint32[] memory operatorSetIdsToCreate, uint32[][] memory operatorSetIds, address[] memory allOperators) {
+        uint256 quorumCount = _registryCoordinator.quorumCount();
+
+        allOperators = new address[](0);
+        operatorSetIdsToCreate = new uint32[](quorumCount);
+
+        // Step 1: Iterate through quorum numbers and get a list of unique operators
+        for (uint8 quorumNumber = 0; quorumNumber < quorumCount; quorumNumber++) {
+            // Step 2: Get operator list for quorum at current block
+            bytes32[] memory operatorIds = _registryCoordinator.indexRegistry().getOperatorListAtBlockNumber(quorumNumber, uint32(block.number));
+        
+            // Step 3: Convert to address list and maintain a sorted array of operators
+            address[] memory operators = new address[](operatorIds.length);
+            for (uint256 i = 0; i < operatorIds.length; i++) {
+                operators[i] = _registryCoordinator.blsApkRegistry().getOperatorFromPubkeyHash(operatorIds[i]);
+                // Insert into sorted array of all operators
+                allOperators = _mergeSortedArrays(allOperators, operators);
+            }
+            address[] memory filteredOperators = new address[](allOperators.length);
+            uint256 count = 0;
+            for (uint256 i = 0; i < allOperators.length; i++) {
+                if (allOperators[i] != address(0)) {
+                    filteredOperators[count++] = allOperators[i];
+                }
+            }
+            // Resize array to remove empty slots
+            assembly { mstore(filteredOperators, count) }
+            allOperators = filteredOperators;
+
+            operatorSetIdsToCreate[quorumNumber] = uint32(quorumNumber);
+        }
+
+        operatorSetIds = new uint32[][](allOperators.length);
+        // Loop through each unique operator to get the quorums they are registered for
+        for (uint256 i = 0; i < allOperators.length; i++) {
+            address operator = allOperators[i];
+            bytes32 operatorId = _registryCoordinator.getOperatorId(operator);
+            uint192 quorumsBitmap = _registryCoordinator.getCurrentQuorumBitmap(operatorId);
+            bytes memory quorumBytesArray = BitmapUtils.bitmapToBytesArray(quorumsBitmap);
+            uint32[] memory quorums = new uint32[](quorumBytesArray.length);
+            for (uint256 j = 0; j < quorumBytesArray.length; j++) {
+                quorums[j] = uint32(uint8(quorumBytesArray[j]));
+            }
+            operatorSetIds[i] = quorums;
+        }
+
+    }
+
+    function _mergeSortedArrays(address[] memory left, address[] memory right) internal pure returns (address[] memory) {
```

Comment:
Nit: potentially can move this to a library?

Reply:
[2ee11df](https://github.com/Layr-Labs/eigenlayer-middleware/pull/286/commits/2ee11dfc9cc92429d9b0b7d75b383e1fb8f7631a)